composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/ui/CreateEventScreen.kt

```diff
@@ -29,19 +39,49 @@ import sp.bvantur.tasky.event.ui.components.TaskyEventType
 import sp.bvantur.tasky.event.ui.components.TaskyReminderPicker
 import sp.bvantur.tasky.event.ui.components.TaskyTimeDatePicker
 import sp.bvantur.tasky.event.ui.components.TaskyVisitorsSection
+import sp.bvantur.tasky.event.ui.model.CreateEventModel
+import sp.bvantur.tasky.event.ui.model.SingleInputModel
 import tasky.composeapp.generated.resources.Res
 import tasky.composeapp.generated.resources.event
-import tasky.composeapp.generated.resources.event_description
-import tasky.composeapp.generated.resources.event_title
 import tasky.composeapp.generated.resources.password_visibility_icon
 
 @Composable
-fun CreateEventRoute(onNavigateBack: () -> Unit) {
-    CreateEventScreen(onNavigateBack = onNavigateBack)
+fun CreateEventRoute(
+    eventModel: CreateEventModel,
+    onNavigateBack: () -> Unit,
+    onOpenSingleInputScreen: (SingleInputModel) -> Unit
+) {
+    val viewModel = koinViewModel<CreateEventViewModel>(
+        parameters = {
+            parametersOf(eventModel)
+        }
+    )
+
+    LaunchedEffect(Unit) {
+        viewModel.onLoadInitialData(eventModel)
+    }
```

Comment:
I see you also use that same model to call a function of the view model 😄 Then the missing constructor declaration of this model makes sense.